central/views/imagecve/types.go

```diff
@@ -27,3 +28,11 @@ type CveView interface {
 	Count(ctx context.Context, q *v1.Query) (int, error)
 	Get(ctx context.Context, q *v1.Query) ([]CveCore, error)
 }
+
+// ResourceCountByCVESeverity is the count of resources affected by cve distributed over severity.
+type ResourceCountByCVESeverity struct {
```

Comment:
A thought for if and when countBySeverity are added for nodeCVE, clusterCVEs. I think this struct could be converted into an interface with `GetCriticalSeverityCount()` etc methods. The `resourceCountBySeverityResolver` can then have this interface as data field. The `imageCVECore.GetImagesBySeverity()` would then return a `ImageCountBySeverity` struct which would implement `ResourceCountBySeverity` . Same for other cveCore types.